File: @Codewars/6_multiples-of-3-or-5/multiples-of-3-or-5.py
"""https://www.codewars.com/kata/514b92a657cdc65150000006/train/python"""

"""v2 - One-linerized version of the same"""
def solution(number):
  return sum({n for n in range(number) if n%3==0 or n%5==0})

"""v1 - Copy of my Project Euler solution, modified to take dynamic max. value"""
# def solution(number):
#   multiples_of_3_or_5 = set()
#   for n in range(number):
#     if n%3==0 or n%5==0:
#       multiples_of_3_or_5.add(n)
#   return sum(multiples_of_3_or_5)

File: @Codewars/6_multiples-of-3-or-5/multiples-of-3-or-5.js
/** https://www.codewars.com/kata/multiples-of-3-or-5/train/javascript */


/** v2 - More readable and efficient */
function solution(number) {
  if (number < 3) {return 0};
  let range = [...Array(number).keys()];
  let multiples = range.filter(num => (num%3 === 0 || num%5 === 0));
  return multiples.reduce(((accumulator, next) => accumulator + next));
}


/** v1 */
// function solution(number) {
//   let range = [...Array((number > 0) ? number : 0).keys()];
//   let multiples = range.filter(num => (num%3 === 0 || num%5 === 0));
//   return multiples.reduce(((accumulator, next) => accumulator + next), 0);
// }
